From patchwork Tue Sep 25 20:57:41 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Jim Mattson <jmattson@google.com>
X-Patchwork-Id: 10614841
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id ACC7513A4
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 25 Sep 2018 20:57:46 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id A03452AA95
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 25 Sep 2018 20:57:46 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 9443F2AD76; Tue, 25 Sep 2018 20:57:46 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-15.5 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI,
	USER_IN_DEF_DKIM_WL autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 9123A2AA95
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 25 Sep 2018 20:57:45 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726257AbeIZDHI (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 25 Sep 2018 23:07:08 -0400
Received: from mail-qt1-f202.google.com ([209.85.160.202]:52594 "EHLO
        mail-qt1-f202.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1726026AbeIZDHI (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 25 Sep 2018 23:07:08 -0400
Received: by mail-qt1-f202.google.com with SMTP id s1-v6so9918098qte.19
        for <kvm@vger.kernel.org>; Tue, 25 Sep 2018 13:57:43 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=google.com; s=20161025;
        h=date:message-id:mime-version:subject:from:to:cc;
        bh=trCCGhnRe84Bz/N/maOYcUT565hxI4RCHyzYD9e4AHM=;
        b=ByzjfMrYXNZOWXQLVOlmVWKKp/4Zhky0vg4y9i1x2VRNIzCpKPrmxkG+T1M9jBPwxl
         XRi+HHOB5CgsTmtFmtpw3s3CINhFx99frmM0l8rwwYGsZP5scgJU8M6LyJmLtRK7mKZ8
         3u1+qBcjiVB9XEHg4Xpr5Gycz236dXWVVauFmHGEBgVjAlOz4Db2gB7ol4PcdV1mBlgY
         pBFb5ll4x0i0+8N6o2Fkenlr87zXU6sulfkNtDPVcMwlkHXpiG2Kqb0dg5L86e0WToeu
         FgWFXh1QSK47Cm0+CIp198dCBzuJr8AvXktsDkTMZ720Xef87F2ptD7m5/RvwrHQOjbh
         wMXQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:date:message-id:mime-version:subject:from:to:cc;
        bh=trCCGhnRe84Bz/N/maOYcUT565hxI4RCHyzYD9e4AHM=;
        b=NN3f+hqc3Ufs49r0Ihiyq15eysRQzNre4f2ASsLDCLN04wLTuxHq9x/+4+vmoQO3hl
         bU7KFow/73Bup4e7dL60X9K9dK0V9xrA359m5gvIGcpxvuYZaSuQVu3CZhJWMvBri0GT
         gJIFEwanXJwM9bxHUj+mBKvgwQDVYDZidtn0UrjE68tbccLBKqcluziEa8zdsQSuuwJY
         Rpa2RsQx69Q68bHB8pdUkmksR9jHXPiIIoRQSfkIAAou210Wo7/GFJKkHpBtIkhA/xoj
         /Gcrd7UiykQN+pDjXGzh7F1BsifVEhY4SNiGVN5dt5Z1PBVLI2IpOgj9H9LHs/L9QSs3
         Na5w==
X-Gm-Message-State: ABuFfogeGp/FEfNybPd/ScJUVXbtG/alE9CO9djq6rWzIysOPecClI0K
        O3i/sQ2m05+bT4OCBzh5YEV/mVknve9ZNDUDPCMEbweVPn3NWDypGmICV9GUFNWNC16zSVie9bh
        zwB6vVIDBJB4janzSW7pAY7wUmVyn3PLJTMaEuvIVh8TwLrWWpk+F47gXKg0b3JE=
X-Google-Smtp-Source: 
 ACcGV63d+WEgJBrxtgXuiHlXVg9YEQmm19Y55yARrkuInk5MWZdN9y5GNZEwwOfuClzB3QgZYOiCl7eOuXbd+w==
X-Received: by 2002:ac8:2f8f:: with SMTP id
 l15-v6mr987386qta.49.1537909062799;
 Tue, 25 Sep 2018 13:57:42 -0700 (PDT)
Date: Tue, 25 Sep 2018 13:57:41 -0700
Message-Id: <20180925205741.162254-1-jmattson@google.com>
Mime-Version: 1.0
X-Mailer: git-send-email 2.19.0.605.g01d371f741-goog
Subject: [PATCH] kvm: vmx: Remove a mysterious FIXME
From: Jim Mattson <jmattson@google.com>
To: kvm@vger.kernel.org
Cc: Jim Mattson <jmattson@google.com>
Content-Type: text/plain; charset="UTF-8"
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

Bits 63:32 of %rax and %rdx are cleared in handle_rdmsr, as they
should be.  There is nothing here to fix.

Signed-off-by: Jim Mattson <jmattson@google.com>
---
 arch/x86/kvm/vmx.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/x86/kvm/vmx.c b/arch/x86/kvm/vmx.c
index 06412ba46aa3..aba4780fbe82 100644
--- a/arch/x86/kvm/vmx.c
+++ b/arch/x86/kvm/vmx.c
@@ -7452,7 +7452,6 @@ static int handle_rdmsr(struct kvm_vcpu *vcpu)
 
 	trace_kvm_msr_read(ecx, msr_info.data);
 
-	/* FIXME: handling of bits 32:63 of rax, rdx */
 	vcpu->arch.regs[VCPU_REGS_RAX] = msr_info.data & -1u;
 	vcpu->arch.regs[VCPU_REGS_RDX] = (msr_info.data >> 32) & -1u;
 	return kvm_skip_emulated_instruction(vcpu);
